ppa/archive/management/commands/gensim_serialize.py

```diff
@@ -0,0 +1,292 @@
+"""
+**gensim_serialize** is a custom manage command to generate and serialize
+a Gensim corpus from Solr.  It should be run *after* content has been indexed
+into Solr via the **index** manage command.
+
+The Corpus is serialized in the Matrix Market format (https://math.nist.gov/MatrixMarket/formats.html)
+with a .mm extension, using the Gensim Topic Modelling library (https://radimrehurek.com/gensim)
+Typically, an index corresponding to the .mm file is also saved by Gensim, with a .mm.index extension.
+
+A dictionary corresponding to token IDs is also saved by default, using a .mm.dict extension.
+This is a utf8-encoded and newline-separated file where the line number N contains the token with word_id N-1.
+Saving the dictionary can be skipped by using the --no-dictionary option.
+
+Additional document-level metadata found in the Solr Index is also saved by default, with .mm.metadata extension.
+This is a csv file with a header row and one row per unique document found in the Solr index.
+Saving the metadata can be skipped by using the --no-metadata option.
+
+By default, _all_ documents found in the Solr index are serialized. This can be controlled using (one of):
+  --doc-id: Document ID(s) to serialize. Multiple documents can be specified by multiple --doc-id flags.
+  --doc-limit: Max no. of documents to serialize.
+
+For corpus generation, the following preprocessing options are available via the --preprocess flag:
+
+    'lower' = Lower-cases words
```

Comment:
Given that corpus generation is a time and space consuming process, I do think it's important to have *some* preprocessing options available at this initial stage of corpus generation, even if they are rarely used.

If we want to trim the no. of preprocessing flags, I suggest we at least keep `lower/strip_punctuation` since I can't imagine a useful corpus without this preprocessing (and we'd be just forcing the user to find other ways to do this downstream).

As an FYI, without preprocessing, the `ppa` corpus is 1.6GB, with dictionary size 247MB. With lower/strip_punctuation, it is 900MB and 90MB respectively.

How about we keep them for the next few months (if for nothing than just our own testing), and take them out later if needed?

Reply:
That makes a lot of sense to me - and good to know about the impact on size!  I wonder if we will want to provide a couple of different versions of the corpus for people to use depending on what their goals are?

My sense is that we don't know enough yet about which options will be best for this corpus; I'm fine with leaving them as is while we continue to work with it.